### ISSUE TITLE
precision 0

### ISSUE THREAD

**Issue by petrabarus** [kartik-v/yii2-money#9]
It's rarely we use decimal in my country. So when I tried

``` php
 <?=
        $form->field($model, 'defaultPricePerUnit')->widget(\kartik\money\MaskMoney::class,
                [
            'pluginOptions' => [
                'prefix' => 'Rp.',
                'suffix' => '',
                'thousands' => '.',
                'decimal' => ',',
                'affixesStay' => true,
                'precision' => 0,
                'allowZero' => true,
                'allowNegative' => false
            ]
        ]);
?>
```

I tried inputting `123456789`, in the textfield it's `Rp.123.456.789`, but in the hidden field it's somewhat `123456.789`. 
Did I put wrong config?

**Comment by gharach** [comment]
i have a same problem with precision 0

**Comment by gael-wogenstahl** [comment]
Hello !
Same issue here but it's not related to @kartik-v extension.
The problem stands in jquery-maskmoney plugin => see ![this issue](https://github.com/plentz/jquery-maskmoney/issues/123)
It seems however that things aren't moving so much on that repository. There are lots of open issues and pull requests waiting...
Will try to find another option for my money input without decimal and post it here.

**Comment by ShadesOB** [comment]
I had same issue. Looked into it and it appears the issue is with the js unmasked function.
It expects that there will be a decimal at the end of the value. 
A fix (maybe not best) is to wrap the decimalPart split with a check.
i.e.
<pre>
if(value.indexOf(".") !== -1){
    $(value.split(/\D/).reverse()).each(function (index, element) {
        if(element) {
            decimalPart = element;
            return false;
       }
    });
}
</pre>

**Comment by ShadesOB** [comment]
This is just a work around. But if your precision is "0", after you output the field you can add js like in the demo below.

<pre>
$field = $form->field($model, '???');
echo $field->widget(MaskMoney::classname(), [
    'pluginOptions' => [
		'precision' => 0,
		...
    ]
]);

$this->registerJs('
	$("[name=\''.$field->model->formName().'['.$field->attribute.']'.'\']").on("change", function () {
	    var val = $(this).val()
	    while(parseInt(val) != val){
	        val *= 10;
	    }
	    $(this).val(val);
	});
');
</pre>

It's not an amazing solution. But it's something.

**Comment by petrabarus** [comment]
@ShadesOB: It's good work around actually. :+1: :+1: :+1:

**Comment by ShadesOB** [comment]
I realised today that this issue also affects loading the page with data. As in what the value is when loading the page.
Kartik does:
<pre>
.maskMoney('mask', val);
</pre>

which doesn't work when you have precision 0;
Instead you have to do:
<pre>
.val(val);
.maskMoney("mask");
</pre>

So an update version of my last post would be
<pre>
$this->registerJs('
	var input = $("[name=\''.$field->model->formName().'['.$field->attribute.']'.'\']");
	input.on("change", function () {
		var val = $(this).siblings("input").val();
		val = Number(val.replace(/[^0-9\.]+/g,""));
	    $(this).val(val);
	});

	setTimeout(function(){							
		var input = $("[name=\''.$field->model->formName().'['.$field->attribute.']'.'\']");								
		input.siblings("input").val('.$model->$attribute.');
		input.siblings("input").maskMoney("mask");
	},1);
');
</pre>

As a side question. If anyone can tell me how js scopes work in this example I would appreciate it.
I have the "var input = " in there twice as it didn't inherit the input variable. :s

**Comment by anish-adm** [comment]
@ShadesOB , Thanks you very much! It worked like a charm :+1:

**Comment by wilsonxyz** [comment]
hi, i try to implement @ShadesOB solution, but i still don't understand,

please advice. how to implement to my code.
Thanks before


```
<?= $form->field($model, 'bruto')->widget(kartik\money\MaskMoney::className(),
            [				
                'pluginOptions' => [
                'suffix' => 'Kg',	
				'precision' => 0,
                ]
]) ?>
$('#poagen-bruto').change(function(){
	calculateNetto();
});`
$('#poagen-tarra').change(function(){
	calculateNetto();
});
function calculateNetto(){	
	var bruto = $('#poagen-bruto').val();			
	var tarra = $('#poagen-tarra').val();	
	$('#poagen-netto').val(bruto-tarra);
};
```

**Comment by ShadesOB** [comment]
First up, apologies as I'm on a new (non yii) project and can't test any of
this. Also it's been some time. :p

There are two problems when helping out with an answer.
1) Your code has two inputs I know nothing about   i.e. "tarra" and "netto"
2) You didn't actually say what's wrong. Is it simply that it's working but
you don't understand why?
Is it that bruto \ tarra are displaying wrong (after change or on page load)
Is it that bruto \ tarra values are wrong
Is it that netto isn't updating? (after change or on page load)

But I would try changing:

$('#poagen-netto').val(bruto-tarra);   =>
$('#poagen-netto').val(bruto-tarra).trigger("change");

Or doing:

$('#poagen-bruto,#poagen-tarra,#poagen-netto').on("change", function
() {        var val = $(this).siblings("input").val();        val =
Number(val.replace(/[^0-9\.]+/g,""));        $(this).val(val);    });

But again. I'm not in a position to test this. So only guessing.

**Comment by kartik-v** [comment]
Resolved via [updates & enhancements to release v1.2.2](https://github.com/kartik-v/yii2-money/commit/6c18c47b6561e52ecd4cff9d2e0c3a0e974c236f). Duplicate to #26, #23, #22, #16.
